lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,369 @@
+# Copyright (c) 2018 Deric Crago <deric.crago@gmail.com>
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+import re
+from os.path import dirname, exists, getsize
+from socket import gaierror
+from ssl import SSLEOFError, SSLError
+from time import sleep
+
+import requests
+import urllib3
+
+from ansible.errors import AnsibleError, AnsibleFileNotFound
+from ansible.module_utils._text import to_bytes, to_native
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.path import makedirs_safe
+
+try:
+    from pyVim.connect import Disconnect, SmartConnect, SmartConnectNoSSL
+    from pyVmomi import vim
+
+    HAS_PYVMOMI = True
+except ImportError as e:
+    HAS_PYVMOMI = False
+    PYVMOMI_IMPORT_ERROR = e
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    author: Deric Crago <deric.crago@gmail.com>
+    connection: vmware_tools
+    short_description: Execute modules via VMware Tools.
+    description:
+      - Execute modules via VMware Tools.
+      - "Note: Windows VMs will need to have C(ansible_shell_type: powershell) set."
+    version_added: "2.8"
+    requirements:
+      - pyvmomi (python library)
+    options:
+      connection_address:
+        description:
+          - Address for the connection (vCenter or ESXi Host).
+        vars:
+          - name: ansible_vmware_tools_connection_address
```

Comment:
```suggestion
          - name: ansible_vmware_tools_host
```

Reply:
would be good to also support the env vars from the vmware inventory plugin: https://docs.ansible.com/ansible/latest/plugins/inventory/vmware_vm_inventory.html

or better yet, the env vars specified by VMware: https://code.vmware.com/docs/6536/vsphere-sdk-for-perl-programming-guide/doc/GUID-66555F76-570A-4C76-BEA7-6C371BF685D6.html

I could also forsee a 'vmware' connection plugin that could be used with the vmware modules, for which I'd anticipate its corresponding var to be `ansible_vmware_host`, which I'd proactively use here.